specfile/specfile.py

```diff
@@ -179,11 +258,26 @@ def rpm_spec(self) -> rpm.spec:
 
     def reload(self) -> None:
         """Reloads the spec file content."""
-        self._lines, self._trailing_newline = self._read_lines(self.path)
+        try:
+            if isinstance(self._file, IOBase) and hasattr(self._file, "name"):
+                file_path = Path(self._file.name)
+                if file_path.exists():
+                    self._file.close()
+                    self._file = file_path.open("r+", **ENCODING_ARGS)
+            self._lines, self._trailing_newline = self._read_lines(self._file)
+        except AttributeError:
+            self._lines, self._trailing_newline = self._read_lines(self._file)
```

Comment:
There are quite some unnecessary checks - `isinstance(self._file, IOBase)` is always true, `hasattr(self._file, "name")` is covered by the `try`-`except`. As for checking the existence, it doesn't make much sense, if the path doesn't exist anymore we do nothing?

```suggestion
        try:
            path = Path(self._file.name)
        except AttributeError:
            pass
        else:
            # reopen the file, in case it has been replaced
            self._file.close()
            self._file = path.open("r+", **ENCODING_ARGS)
        self._lines, self._trailing_newline = self._read_lines(self._file)
```

Reply:
The file.exists() check was added to ensure the previously failing cockpit test passes. If the path no longer exists and cannot be reopened, the existing file content is retained. Without this check, _file, self._lines, and self._trailing_newline would never be initialized, preventing _read_lines(self._file) from executing.
I do wonder if this was even remotely a valid fix.

Reply:
I wonder, is it the same test I have fixed [here](https://github.com/packit/packit/pull/2532/files)? The PR is blocked on something else so it's not merged yet, but if the fix works in this case we can merge it separately.

Reply:
The previous implementation did not reopen any file handles and simply read again from the already opened file.
To preserve that behaviour, I just read the already open file again and do nothing.
What do you think if it was fine to do so ? I am a bit puzzled.

Reply:
> The previous implementation did not reopen any file handles and simply read again from the already opened file.

Yes, and that didn't match the original behavior so it was changed. There should be no exceptions to that because of a wrongly written test (if that's the case).

Reply:
Thinking about it, [this](https://github.com/packit/packit/pull/2564) could also fix the test (and is probably a reason why it worked in the first place, though I still think it is badly written).